action.yml

```diff
@@ -36,3 +39,16 @@ runs:
         git config --global --add credential.helper store
         echo $'protocol=https\nhost=github.com\nusername=x-access-token\npassword='"${GH_FEDERATION_ACCESS_TOKEN}"$'\n' | \
           git credential-store store
+
+    - name: Store access token for repos as extraheader
+      if: inputs.repos != ''
+      shell: bash
+      env:
+        REPOS: ${{ inputs.repos }}
+      run: |
+        TOKEN=$(echo -n "x-access-token:${GH_FEDERATION_ACCESS_TOKEN}" | base64)
+        echo "::add-mask::${TOKEN}"
+        for repo in ${REPOS}; do
+          echo "config extraheader for ${repo}"
+          git config --global "http.https://github.com/${repo}.extraheader" "Authorization: Basic ${TOKEN}"
```

Comment:
It's better to respect GITHUB_SERVER_URL in order to support GitHub Enterprise Servers. 
ref: https://github.com/actions/checkout/blob/96f53100ba2a5449eb71d2e6604bbcd94b9449b5/src/url-helper.ts#L26C48-L26C48